Add chi2 to jax.scipy.stats

Dpananos opened the pull request:
Hello,

I realized the chisquare distribution was missing from `jax.scipy.stats` so I went ahead and added it.  I've also added some tests as well (they all pass when I run `python tests/scipy_stats_test.py --test_targets="LaxBackedScipyStatsTests.testChi2" --num_generated_cases=100`).

jakevdp reviewed (changes requested):
This looks great! Thanks for the contribution. One remaining piece is to add this new functionality to the documentation here: https://github.com/google/jax/blob/master/docs/jax.scipy.rst

Finally, if you're able to do so, please squash everything into a single commit and force-push to the branch. Thanks!

jakevdp commented:
Also, please mention this new function in the changelog here: https://github.com/google/jax/blob/master/docs/CHANGELOG.rst

jakevdp commented on docs/CHANGELOG.rst:
```diff
@@ -11,6 +11,7 @@ jax 0.2.10 (Unreleased)
 -----------------------
 * `GitHub commits <https://github.com/google/jax/compare/jax-v0.2.9...master>`__.
 * New features:
+  * :func: `jax.scipy.stats.chi2` is now available as a distribution with logpdf and pdf methods.
```
This will not render correctly with the space. Please change to
```
:func:`jax.scipy.stats.chi2`
```

Dpananos replied:
Will do, thanks!

Dpananos committed:
Add logdf and pdf for chisquare distribution

Add tests

Lint with flake8 fails.  Should pass now

newline at end of file for flake8

docs and changes

remove whitespace in changeloc

Dpananos commented:
OK, I think I properly squashed the commits.  Let me know.  If not, I think I may have to burn this branch and try again (sorry, my git is still not great :/)

jakevdp reviewed (approved):
Looks great, thanks!

jakevdp commented:
Looks like GPU & TPU tests pass as well... we'll get this merged in a bit. Thanks for the great contribution!

Dpananos commented:
Is there interest in adding more distributions?  This was a fun distraction from other stuff.  Mind if I add a few more?

jakevdp commented:
Sure, feel free to add more!
